microsoft/PowerToys
# [DSC]Improve module generation for versioned release install

## Summary of the Pull Request

Fixes some issues related to using DSC after testing on a versioned release install.
- Also generate .psd1 file during build to set the version of the manifest to PowerToys built version instead of having a hardcoded 0.0.1.
- Don't expose internal CI server build paths from release builds on .psm1 file.
- Use -Force to stop the PowerToys process when applying configurations. This is required for stopping the initial process that runs right after install.


## Validation Steps Performed
- Built a local installer of PowerToys after setting Version.props version to `0.80.143`.
- Installed 0.80.143 and let PowerToys run its first time.
- In PowerShell 7, ran `winget configure .\disableAllModules.dsc.yaml`  ( https://github.com/microsoft/PowerToys/blob/7b89482b944eb3845f0445a5c1f1fe7f983d5f96/src/dsc/Microsoft.PowerToys.Configure/examples/disableAllModules.dsc.yaml )
- After the configuration process is done, verify that PowerToys has restarted with all its modules disabled.
- Verified that the installed .psd1 file ( in "C:\Program Files\WindowsPowerShell\Modules\Microsoft.PowerToys.Configure\0.80.143" for machine installs) contains the `0.80.143` version in the manifest)
- Verified that the function `GetPowerToysSettingsPath` inside the installed .psm1 file doesn't contain a local build path for debugging processes.
